pallets/process-validation/src/lib.rs

```diff
@@ -79,50 +100,113 @@ pub mod pallet {
 
     /// Storage map definition
     #[pallet::storage]
-    #[pallet::getter(fn processes_by_id_and_version)]
-    pub(super) type ProcessesByIdAndVersion<T: Config> = StorageDoubleMap<
+    #[pallet::getter(fn get_process)]
+    pub(super) type GetProcess<T: Config> = StorageDoubleMap<
         _,
         Blake2_128Concat,
         T::ProcessIdentifier,
         Blake2_128Concat,
-        T::ProcessVersion,
+        i32,
         Process,
+        ValueQuery,
+    >;
+
+    #[pallet::storage]
+    #[pallet::getter(fn get_version)]
+    pub(super) type GetVersion<T: Config> = StorageMap<
+        _,
+        Blake2_128Concat,
+        T::ProcessIdentifier,
+        i32,
         ValueQuery
     >;
 
     #[pallet::event]
     #[pallet::generate_deposit(pub(super) fn deposit_event)]
     pub enum Event<T: Config> {
-        // TODO: implement correct events for extrinsics including params
-        ProcessCreated,
-        ProcessDisabled
+        // id, version, restrictions, is_new
+        ProcessCreated(T::ProcessIdentifier, i32, Vec<Restriction>, bool),
+        ProcessDisabled,
     }
 
     #[pallet::error]
     pub enum Error<T> {
         // TODO: implement errors for extrinsics
     }
 
+    // helper functions
+    fn get_version<T: Config>(id: T::ProcessIdentifier) -> i32 {
+        return <GetVersion<T>>::get(&id);
+    }
+
+    fn uopdate_version<T: Config>(id: T::ProcessIdentifier) -> i32 {
+        let version: i32 = get_version::<T>(id.clone());
+        let exists: Result<Process, ()> = <GetProcess<T>>::try_get(id.clone(),version);
```

Comment:
This check here I'm uncertain about. If this check fails we have a data integrity problem. You could probably just skip this (as it doens't help us see or solve the integrity problem anyway)

Reply:
Was thinking that this could be handled by the pallet::error? Still to be implemented, plan for today, there is a little toido in the descriptionj

Reply:
As discussed in slack the question really is what is the error condition? I don't think the previous version not existing is a huge problem and shouldn't error. What might be worth an error though is if the new proposed version already exists. That would be really bad. Not sure how we'll test that behaviour though 🤔

Reply:
Replaced with `contains_key` which returns a boolean value. I guess it makes more sense, pushing shortly!